### ISSUE TITLE
LM training data

### ISSUE THREAD

**Issue by cri5Castro** [facebookresearch/wav2letter#521]
HI there I'm working in model for spanish using w2l  unfortunately due to the small size of my dataset the training data is not big enough to train a robust language model (i'm using the GCNN fairseq approach) so i was wondering if is possible to train the language model with different data that the AM for example using a big spanish corpus using the same preprocessing as the AM training data of course

**Comment by lunixbochs** [comment]
Yes, you can train an LM separately as long as you use the same tokens

**Comment by cri5Castro** [comment]
thanks for the nice help , im working in the common voice dataset ;)

**Comment by tlikhomanenko** [comment]
@cri5Castro,

I would also recommend to test LM trained on large corpus (this definitely will be out of domain) on your domain small data and compare with the one you train on domain data (you can compare with 4gram model and with fairseq trained with high dropout, like 0.3-0.4). Your model trained on large data could poorly preform on in-domain data compared to the one you trained on small in-domain data. Then you can do several epoch of finetuning this large model on in-domain data.

Will close the issue, feel free to reopen if further discussion is needed.
